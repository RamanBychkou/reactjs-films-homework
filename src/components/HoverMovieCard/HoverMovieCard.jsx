import React from 'react';
import style from './HoverMovieCard.scss';
import * as stylerate from '../Star/Star.scss';
import * as styleBtn from '../FilmButtons/FilmButtons.scss';

const showOverwie = (e) => {
  let element = e.target.parentElement;
  element.classList.add(style.hoverControlNone);
  element.nextSibling.classList.add(style.hoverOverwieBlock);
  while (element.localName !== 'article') {
    element = element.parentElement;
  }
  element.querySelector('figure').classList.add(style.figcaptionNone);
};

const returnHover = (e) => {
  const element = e.currentTarget;
  if (element.tagName === 'DIV') {
    element.classList.remove(style.hoverOverwieBlock);
    element.previousSibling.classList.remove(style.hoverControlNone);
  } else {
    element.parentElement.classList.remove(style.hoverOverwieBlock);
    element.parentElement.previousSibling.classList.remove(style.hoverControlNone);
  }
  document.querySelector(`.${style.figcaptionNone}`).classList.remove(style.figcaptionNone);
};

const hoverMovieCard = (props) => {
  const { 
genres, overview, title, rate 
} = props;
  return (
    <div className={style.hoverMovie}>
      <div className={style.hoverControl}>
        <label htmlFor="playTrailer">
          <input type="button" className={style.playTrailer} name="playTrailer" id="playTrailer" />
                  Watch Now
        </label>
        <button type="button" className={style.showOverwie} onClick={showOverwie}>View Info</button>
      </div>
      <div className={style.hoverOverwie} onMouseLeave={returnHover}>
        <button type="button" className={style.hoverClose} onClick={returnHover}>&#9587;</button>
        <div className={style.hoverOverwieTitle}>
          <div className={style.hoverOverwieTitleItem}>
            <h3>{title}</h3>
            <ul>
              {genres}
            </ul>
          </div>
          <div className={style.hoverOverwieTitleItem}>
            <p className={stylerate.rate}>{rate}</p>
          </div>
        </div>
        <p>{overview}</p>
        <button type="button" className={styleBtn.watch}>Watch Now</button>
      </div>
    </div>
  );
};

export default hoverMovieCard;
