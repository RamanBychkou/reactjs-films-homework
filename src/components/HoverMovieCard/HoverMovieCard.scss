.hoverMovie {
  position: absolute;
  top: 0;
  display: none;
  padding: 0 1rem;
  width: 90%;
  height: 80%;
  color: #fff;
  background: rgba(0, 0, 0, 0.64);

  input,
  button {
    outline: none;
  }

  label {
    display: flex;
    justify-content: center;
    align-items: center;
    flex-flow: column nowrap;
    position: relative;
    margin-bottom: 1rem;
    cursor: pointer;

    .playTrailer {
      border-radius: 20rem;
      width: 7rem;
      height: 7rem;
      border: none;
      background-color: #3bb3df;
      margin-bottom: 1rem;
    }

    &::before {
      content: "";
      position: absolute;
      border: 1.7rem solid transparent;
      border-left: 2.5rem solid #fff;
      top: 1.9rem;
      left: 2.7rem;
    }
  }

  .showOverwie {
    background: none;
    border-radius: 5px;
    border: 1px solid #fff;
    color: #fff;
    padding: 0.4rem 1rem;
    cursor: pointer;
  }

  .hoverClose {
    background: none;
    color: #fff;
    border: none;
    cursor: pointer;
    outline: none;
  }

  .hoverOverwieTitleItem {
    &:first-child {
      margin-right: 0.5rem;
      width: 15rem;
    }

    h3 {
      overflow: hidden;
      text-overflow: ellipsis;
      text-transform: uppercase;
      margin-bottom: 0.3rem;
      white-space: nowrap;

      &:hover {
        white-space: normal;
        text-overflow: initial;
        word-break: keep-all;
      }
    }

    ul {
      display: flex;
      justify-content: flex-start;
      flex-flow: row wrap;
      color: #47c3f0;

      span {
        cursor: pointer;
      }
    }
  }
}

article:hover .hoverMovie {
  display: flex;
  justify-content: center;
  align-items: center;
  flex-flow: column nowrap;
}

.hoverOverwie {
  display: none;
  min-width: 100%;
  top: 0;
  left: 0;
  height: 100%;
}

.hoverControlNone {
  display: none;
}

.hoverOverwieBlock {
  display: flex;
  align-items: flex-start;
  flex-flow: column nowrap;
  justify-content: center;
  height: 100%;
  word-break: keep-all;

  .hoverOverwieTitle {
    padding: 1rem 0;
    display: flex;
    flex-flow: row nowrap;
    justify-content: space-between;
    align-items: baseline;
    word-wrap: break-word;
  }

  p {
    margin-bottom: 1rem;
  }
}

.hoverControlBlock {
  text-align: center;
}
