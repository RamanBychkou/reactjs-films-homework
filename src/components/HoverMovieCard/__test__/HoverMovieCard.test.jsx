import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import HoverMovieCard from '../index';

test('HoverMovieCard renders correctly', () => {
  const renderer = new ShallowRenderer();
  renderer.render(<HoverMovieCard />);
  const result = renderer.render(<HoverMovieCard />);
  expect(result).toMatchSnapshot();
});
