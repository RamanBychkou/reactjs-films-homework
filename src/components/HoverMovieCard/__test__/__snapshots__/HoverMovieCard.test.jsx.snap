// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HoverMovieCard renders correctly 1`] = `
<div
  className="hoverMovie"
>
  <div
    className="hoverControl"
  >
    <label
      htmlFor="playTrailer"
    >
      <input
        className="playTrailer"
        id="playTrailer"
        name="playTrailer"
        type="button"
      />
      Watch Now
    </label>
    <button
      className="showOverwie"
      onClick={[Function]}
      type="button"
    >
      View Info
    </button>
  </div>
  <div
    className="hoverOverwie"
    onMouseLeave={[Function]}
  >
    <button
      className="hoverClose"
      onClick={[Function]}
      type="button"
    >
      â•³
    </button>
    <div
      className="hoverOverwieTitle"
    >
      <div
        className="hoverOverwieTitleItem"
      >
        <h3 />
        <ul />
      </div>
      <div
        className="hoverOverwieTitleItem"
      >
        <p />
      </div>
    </div>
    <p />
    <button
      type="button"
    >
      Watch Now
    </button>
  </div>
</div>
`;
